docs/source/advanced/response-based.mdx

```diff
@@ -1,18 +1,18 @@
 ---
-title: Code generation methods in Apollo Kotlin
+title: Response based codegen
 ---
 
-Apollo Kotlin provides multiple code generation engines (**codegens**) that you can choose from depending on your use case:
+Apollo Kotlin takes your GraphQL operations, generates Kotlin models for them and instantiates them from Json response allowing you to access your data in a type safe way. There are 3 different domains at play:
 
+* The GraphQL domain: operations
+* The Kotlin domain: models
+* The JSON domain: responses
 
-| Codegen | Description |
-|---------|-------------|
-| `operationBased` (default) | Generates models according to the shape of each defined operation. |
-| `responseBased` | <p>Generates models according to the shape of each operation's response.</p><p>Compared to `operationBased`, models generated by the `responseBased` codegen are more performant and expose more type information, at the cost of generating more total code. [See details.](#the-responsebased-codegen)  </p> |
-| `compat` (deprecated) | Similar to `operationBased`, except it generates models that are compatible with Apollo Android v2. This codegen helps projects upgrade from v2 to v3. |
+By default, Apollo Kotlin generates models that match 1:1 with your GraphQL operations. Inline and named fragments generate synthetic fields, so you can access GraphQL fragments with Kotlin code like `data.hero.onDroid.primaryFunction`. This code generation engine (**codegen**) is named `operationBased` because it matches the GraphQL operation.
 
-> 💡 For more in-depth information on each codegen, see [this design document](https://github.com/apollographql/apollo-android/blob/main/design-docs/Codegen.md).
+The Json response may have a different shape than your GraphQL operation though. This is the case when using merged fields or fragments. If you want to access your Kotlin properties as they are in the JSON response, Apollo Kotlin provides a `responseBased` codegen that matches the JSON response. GraphQL fragments are represented as Kotlin interfaces, so you can access their fields with Kotlin code like `(data.hero as Droid).primatryFunction`. Because it maps to the JSON response, the `responseBased` models have the property of allowing JSON streaming and/or mapping to dynamic JS objects. But because GraphQL is a very expressive language, [it's also easy to create a GraphQL query that generate a very large JSON response](https://github.com/apollographql/apollo-kotlin/issues/3144). **For this reason and other [limitations](#responsebased-codegen-limitations), we recommend using `operationBased` codegen by default**.
```

Comment:
```suggestion
The Json response may have a different shape than your GraphQL operation though. This is the case when using merged fields or fragments. If you want to access your Kotlin properties as they are in the JSON response, Apollo Kotlin provides a `responseBased` codegen that matches the JSON response. GraphQL fragments are represented as Kotlin interfaces, so you can access their fields with Kotlin code like `(data.hero as Droid).primaryFunction`. Because it maps to the JSON response, the `responseBased` models have the property of allowing JSON streaming and/or mapping to dynamic JS objects. But because GraphQL is a very expressive language, [it's also easy to create a GraphQL query that generate a very large JSON response](https://github.com/apollographql/apollo-kotlin/issues/3144). **For this reason and other [limitations](#responsebased-codegen-limitations), we recommend using `operationBased` codegen by default**.
```